app/Models/Role.php

```diff
@@ -145,7 +160,7 @@ public static function boot()
             if ($relationName === 'users') {
                 foreach ($pivotIds as $pivotId) {
                     $user = User::find($pivotId);
-                    activity()->causedBy(auth()->user())->performedOn($user)
+                    activity()->causedBy($user)->performedOn($user)
```

Comment:
`$user` is overloaded here. it's already defined on line 162

Reply:
Should be fixed in latest. I also added some safeguards to prevent server errors in case these code paths are entered from side effects (ie: a command, Horizon job, etc).

Reply:
I see you removed the assignment of `$user=Auth::user()`, but this statement previously was `causedBy(auth()->user())` and now it's `causedBy(User::find($pivotId))`.

The same change occurs a few lines up (151).